File: algorithm-pattern-problems/c/array_and_hashing/remove_element.c
int removeElement(int* nums, int numsSize, int val) {
  int k = 0;

  for (int i = 0; i < numsSize; i++){
    if (nums[i] == val) {
      continue;
    }
    // swap
    int temp = nums[i];
    nums[i] = nums[k];
    nums[k] = temp;
    k++;
  }

  return k;
  // Time: O(n)
  // Space: O(1)
}

File: algorithm-pattern-problems/golang/array_and_hashing/remove_element.go
package arrayandhashing

func removeElement(nums []int, val int) int {
	k := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			// swap
			nums[i], nums[k] = nums[k], nums[i]
			k++
		}
	}

	return k
	// Time: O(n)
	// Space: O(1)
}
